runtime/syntax/sml.vim

```diff
@@ -152,9 +153,9 @@ syn match    smlKeyChar      ";"
 syn match    smlKeyChar      "\*"
 syn match    smlKeyChar      "="
 
-syn match    smlNumber        "\<-\=\d\+\>"
-syn match    smlNumber        "\<-\=0[x|X]\x\+\>"
-syn match    smlReal          "\<-\=\d\+\.\d*\([eE][-+]\=\d\+\)\=[fl]\=\>"
+syn match    smlNumber        "\<\~\=\%(0w\)\=\d\+\>"
+syn match    smlNumber        "\<\~\=0w\=x\x\+\>"
+syn match    smlReal          "\<\~\=\d\+\.\d*\([eE][-+]\=\d\+\)\=[fl]\=\>"
```

Comment:
```suggestion
syn match    smlNumber        "\~\=\<0w\=x\x\+\>"
syn match    smlReal          "\~\=\<\d\+\.\d*\([eE]\~\=\d\+\)\=[fl]\=\>"
```

`~` is not in `'iskeyword'` so the word boundary won't work there. `+` is not supported in the exponent and `~` should also be used there.